### PR TITLE
cache nsurlsession info of sample app and its extension in app group …

### PR DESCRIPTION
**Pull request by thuyBox**
…and updates to run sample app on device

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @thuyBox has signed the CLA. Thanks for the pull request!

File: BoxContentSDKSampleApp/shareExtension/ShareViewController.m [hunk 1]
```diff
@@ -61,6 +65,14 @@ - (void)didSelectPost {
     NSExtensionItem *item = self.extensionContext.inputItems.firstObject;
     NSItemProvider *itemProvider = item.attachments.firstObject;
 
+    if (![itemProvider hasItemConformingToTypeIdentifier:typeIdentifier]) {
+        typeIdentifier = (__bridge_transfer NSString *)kUTTypeMovie;
+
+        if (![itemProvider hasItemConformingToTypeIdentifier:typeIdentifier]) {
+            typeIdentifier = (__bridge_transfer NSString *)kUTTypeZipArchive;
```
Comment (adempsey) [inline BoxContentSDKSampleApp/shareExtension/ShareViewController.m]: Why zip archive?

Comment (thuyBox) [reply BoxContentSDKSampleApp/shareExtension/ShareViewController.m]: if we want to upload bigger object for testing in sample app, usually they are in zip format.

Comment (adempsey) [reply BoxContentSDKSampleApp/shareExtension/ShareViewController.m]: Not guaranteed though, right? Should it be `kUTTypeData` or `kUTTypeItem` or something more generic?

**Review by adempsey — APPROVED** [review]

**Pull request merged**